pennylane/ops/qutrit/matrix_ops.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the discrete-variable quantum operations that
+accept a hermitian or an unitary matrix as a parameter.
+"""
+# pylint:disable=abstract-method,arguments-differ,protected-access
+import warnings
+import numpy as np  # pylint: disable=unused-import
+
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation
+from pennylane.wires import Wires
+
+
+class QutritUnitary(Operation):
+    r"""Apply an arbitrary, fixed unitary matrix.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        U (array[complex]): square unitary matrix
+        wires(Sequence[int] or int): the wire(s) the operation acts on
+    """
+    num_wires = AnyWires
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, *params, wires, do_queue=True):
+        wires = Wires(wires)
+
+        # For pure QutritUnitary operations (not controlled), check that the number
+        # of wires fits the dimensions of the matrix
+        U = params[0]
+
+        dim = 3 ** len(wires)
+
+        if qml.math.shape(U) != (dim, dim):
```

Comment:
It looks like in the updated version of the `QubitUnitary`, there is also a dimension check for a valid batch size; it must be new, we should check with someone whether that needs to be implemented here as well.

Reply:
Yes, I think that is new. It wasn't there when I initially implemented `QutritUnitary`, but it does make sense now as I recall seeing tests for `QubitUnitary` that used a list of unitary matrices as the input parameter for `QubitUnitary`. I'm a bit hesitant to add that feature just yet though because I haven't had a chance to look at the underlying implementation for batches in `QubitUnitary`, and I'm uncertain if it will just work out of the box.

Regardless, I do agree that getting input about whether it needs to be implemented would be helpful.